docs/guide/piece-pages.md

```diff
@@ -1,257 +1,162 @@
+| `currentPage` | A number representing what page of results is shown, starting with `1` |
+| `totalPages` | The total number of results pages there are |
+| `totalPieces` | The total number of pieces across all result pages |
 
-* In A3, there are no tags, so there is no "with these tags" feature to limit what is displayed on a particular piece page. However, you can do this yourself as described later.
+### Pagination
 
-* Infinite scroll and refresh-free filtering don't currently exist in A3. Since A3 is much less opinionated on the front end, they probably won't be part of the core, but they may come back at some point as an optional module.
+```django
+{% import '@apostrophecms/pager:macros.html' as pager with context %}
 
-## Filtering pieces on the Piece Page
+{{ pager.render({
+  page: data.currentPage,
+  total: data.totalPages,
+  class: 'blog-pager'
+}, data.url) }}
+```
 
-Just like in A2, we can configure `piecesFilters` to offer filtering to our website visitors. Let's start by adding a field to our `product` pieces that's good to filter on:
+Index pages do not list every single piece in a single view. That could quickly become a problem. Instead, they include *ten* pieces on `data.pieces` at a time (by default).
 
-```js
-// modules/product/index.js
-module.exports = {
-  // ...
-  fields: {
-    add: {
-      // ... add this as one more field
-      color: {
-        type: 'select',
-        label: 'Color',
-        choices: [
-          {
-            value: 'red',
-            label: 'Red'
-          },
-          {
-            value: 'green',
-            label: 'Green'
-          },
-          {
-            value: 'blue',
-            label: 'Blue'
-          }
-        ]
-      }
-    }
-  }
-};
-```
+The pager macro is a special template using the [Nunjucks macro](https://mozilla.github.io/nunjucks/templating.html#macro) feature. This particular macro accepts two arguments:
 
-```js
-// modules/product-page/index.js
-module.exports = {
-  extend: '@apostrophecms/piece-page-type',
-  options: {
-    label: 'Product Index Page',
-    piecesFilters: [
-      {
-        name: 'color',
-        label: 'Color'
-      }
-    ]
-  }
-};
-```
+- an object with the `currentPage` and `totalPages` values, described above, as well as an optional CSS class for the pager wrapper
+- the page URL, using `data.url`
+
+Other than the class, the pager macro code above will rarely change.
+
+**You can change the number of pieces in each page of results** by setting [the `perPage` option](/reference/module-api/module-options.md#perpage-2) on the module. The data passed to templates will all update, so you don't need to make any other adjustments.
+
+## The show page template
+
+As a reminder, show pages are the web pages for individual pieces, rendered from `show.html` templates. This template uses a very standard page template structure. Assuming our `article` piece type example has a single `main` area, it could look like this:
```

Comment:
```suggestion
Show pages are the web pages for individual pieces, rendered from `show.html` templates. Assuming our `article` piece type example has a single `main` area, it could look like this:
```